azure-pipelines.yml

```diff
@@ -0,0 +1,399 @@
+      ref: refs/tags/v1.0.1
+
+stages:
+- template: stage-with-burgr-notifications.yml@commonTemplates
+  parameters:
+    burgrName: 'build'
+    burgrType: 'build'
+    stageName: 'build'
+    stageDisplayName: Build, UT, analyze on SonarCloud and stage to repox
+    jobs:
+    - job: build
+      displayName: 'Build, UT, analyze on SonarCloud and stage to repox'
+      variables:
+       SIGNTOOL_PATH: 'C:\\Program Files (x86)\\Windows Kits\\10\\bin\\10.0.17763.0\\x64\\signtool.exe'
+       solution: 'SonarScanner.MSBuild.sln'
+       sonarCloudOrganization: 'sonarsource'
+       sonarCloudMsBuildProjectKey: 'sonarscanner-msbuild'
+       sonarCloudMsBuildProjectName: 'SonarScanner for MSBuild'
+       buildPlatform: 'Any CPU'
+       buildConfiguration: 'Release'
+       MAVEN_CACHE_FOLDER: $(Pipeline.Workspace)/.m2/repository
+       MAVEN_OPTS: '-Xmx3072m -Dmaven.repo.local=$(MAVEN_CACHE_FOLDER)'
+       commonMavenArguments: -B -Pdeploy-sonarsource -Dmaven.test.skip=true
+      steps:
+      - checkout: self
+      - task: NuGetToolInstaller@1
+        displayName: 'Use NuGet 4.9.3'
+        inputs:
+          versionSpec: '4.9.3'
+      - task: CacheBeta@0
+        displayName: Cache Maven local repo
+        inputs:
+          key: maven | pom.xml
+          path: $(MAVEN_CACHE_FOLDER)
+      - task: NuGetCommand@2
+        displayName: 'NuGet restore'
+        inputs:
+          command: 'restore'
+          restoreSolution: '$(solution)'
+          feedsToUse: 'select'
+      - task: DownloadSecureFile@1
+        name: snk
+        inputs:
+          secureFile: 'SonarSourceSecret.snk'
+        displayName: 'Download snk'
+      - task: DownloadSecureFile@1
+        displayName: 'Download pfx'
+        name: pfx
+        inputs:
+          secureFile: 'sonarsource-2019-2021.pfx'
+      - task: PowerShell@2
+        displayName: 'Promote project version'
+        inputs:
+          targetType: 'inline'
+          script: |
+            # Calculate the file path
+            $versionFilePath = "$env:BUILD_SOURCESDIRECTORY\scripts\version\Version.props"
+            Write-Host "Reading the Sonar project version from '${versionFilePath}' ..."
+            
+            # Read the version from the file
+            [xml]$versionProps = Get-Content "$versionFilePath"
+            $sonarProjectVersion = $versionProps.Project.PropertyGroup.MainVersion
+            Write-Host "Sonar project version is '${sonarProjectVersion}'"
+            # Set the variable to it can be used by other tasks
+            Write-Host "##vso[task.setvariable variable=SONAR_PROJECT_VERSION]$sonarProjectVersion"
+            Write-Host "Setting versions, sha1 and build number to Version.props file"
+
+            $branchName = $env:BUILD_SOURCEBRANCH
+
+            if ($env:BUILD_SOURCEBRANCH.StartsWith("refs/heads/")) {
+              $branchName = $env:BUILD_SOURCEBRANCH.Substring(11)
+            }
+
+            (Get-Content $versionFilePath) `
```

Comment:
This isn't strictly necessary since you are explicitly passing the properties when building the file in line 101-105.

Reply:
OK i didn't see all the ins and outs of that version.props file, but i'll remove this bit.